### PR TITLE
Override Modern color scheme's highlight color to Blueberry 2

### PR DESCRIPTION
**Pull request by fushar**
Related to: pfsHM7-1qe-p2#comment-1787

## Proposed changes:

We would like to propose updating the Modern color scheme, so that the highlight color, which is present in the admin bar and sidebar, becomes blue (Blueberry 2) instead of green:

|Before|After|
|-|-|
|image|image|
|image|image|

The color is chosen to follow the navbar menu item's active state color in the WordPress.org website:

image

This PR is to have the change faster in dotcom. If we like it, we can propose the change back to dotorg.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

1. Patch this to your site.
2. Go to `wp-admin/profile.php`.
3. Select the Modern scheme and click Update.
4. Verify that the hover text color in the admin bar and sidebar is now blu-ish as per screenshot above.

### CODE REVIEW

**Comment by jasmussen** [comment]
Color can work, sure, I'd like to defer to another developer like Riad on whether this is the right approach.

My instinct is that this is a local hack, that can be okay to have temporarily assuming we contribute the change to WordPress core itself as well. So it might need a core trac patch as well.

**Comment by youknowriad** [comment]
Same thought as @jasmussen Can we make this a core patch?

I was also checking this with @richtabor and he mentioned the possibility of not using a third color (or just reusing the same blue color for second and third colors)

File: projects/packages/masterbar/src/admin-color-schemes/colors/modern/colors.scss
Comment (lsl) [inline projects/packages/masterbar/src/admin-color-schemes/colors/modern/colors.scss]: Is the !important necessary here?

**Review by lsl — DISMISSED** [review]
LGTM

**Comment by fushar** [comment]
> I was also checking this with @richtabor and he mentioned the possibility of not using a third color (or just reusing the same blue color for second and third colors)

@youknowriad @richtabor I tried reusing the second color but it makes the text somehow unreadable on hover due to bad contrast against the black background, see:

I'm not a designer but my gut says to keep the third color, what do you think?

|This PR|Reusing second color|
|-|-|
|image|image|
|image|image|

File: projects/packages/masterbar/src/admin-color-schemes/colors/modern/colors.scss
Comment (fushar) [reply projects/packages/masterbar/src/admin-color-schemes/colors/modern/colors.scss]: I was afraid it loses to Core's, but it turns out it works out just fine, removed!

**Comment by fushar** [comment]
> this is a local hack, that can be okay to have temporarily assuming we contribute the change to WordPress core itself as well. So it might need a core trac patch as well.

> Same thought as @jasmussen Can we make this a core patch?

Yup, I'll make an upstream Core patch after we agree on the colors.

In dotcom, we need to update the color scheme on Calypso as well anyway, so I think it's better to update dotcom first (we can control the deployment timing). Then after Core patch is merged and shipped, we revert the hack.

**Comment by jasmussen** [comment]
> Then after Core patch is merged and shipped, we revert the hack.

Do you have a process for tracking this, so the drift doesn't build up over time?

**Comment by arthur791004** [comment]
> In dotcom, we need to update the color scheme on Calypso as well anyway, so I think it's better to update dotcom first (we can control the deployment timing). Then after Core patch is merged and shipped, we revert the hack.

IMHO it would be better to update Core first to avoid inconsistency because it's not a bug and it might take time to discuss whether it makes sense to change the color of Modern in Core 🤔 People may love to use “Modern” because of the green color...

**Comment by nuriapenya** [comment]
> @youknowriad @richtabor I tried reusing the second color but it makes the text somehow unreadable on hover due to bad contrast against the black background, see. I'm not a designer but my gut says to keep the third color, what do you think?

If we want to reuse the Blueberry, we could consider using it as a background color + white font, as we have it in an active state on the navigation items. It ensures consistency and accessibility, but it's quite heavy on sight as an interaction. So I am not sure 100%, but just as an alternative:

screen 2024-10-14 at 12 29 49

**Comment by fushar** [comment]
> If we want to reuse the Blueberry, we could consider using it as a background color + white font, as we have it in an active state on the navigation items. It ensures consistency and accessibility, but it's quite heavy on sight as an interaction. 

Thanks @nuriapenya for the alternative solution! I just checked all existing color schemes, and I found none of them have this pattern of using the same hover background color as the active menu background color. So, not sure if we want this 🤔

**Comment by fushar** [comment]
Hi folks, I think I just discovered a conclusive solution. So if you are logged in to WordPress.org, you will see the Blueberry 2 (`#7b90ff`) font color as the hover state, which is exactly what is being proposed in this PR:

image

Let's go with this. I guess as the next step I'll propose a patch in Core and stop overriding in dotcom as suggested here pfsHM7-1qe-p2#comment-1790.

**Comment by fushar** [comment]
Added a Core patch: https://core.trac.wordpress.org/ticket/62219

@youknowriad / @jasmussen let me know if this patch is good; I've never landed a patch in Core! 😄 

> Do you have a process for tracking this, so the drift doesn't build up over time?

I'll keep this parent issue:

- https://github.com/Automattic/wp-calypso/issues/95334

open until that patch is accepted, and most likely will just close out this PR soon.

**Comment by youknowriad** [comment]
Patch is great, we might want to wait a few days to land it because I believe we're in beta period for 6.7 and we didn't branch yet (create a dedicated branch for 6.7) which means trunk is closed for enhancements at the moment.

**Pull request closed without merging**
